core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/PathHandler.java

```diff
@@ -0,0 +1,175 @@
+import java.util.Set;
+import java.util.Stack;
+import java.util.stream.Collectors;
+
+public class PathHandler implements OutputOperationHandler<Path, Iterable<Iterable<Edge>>> {
+
+    private final Set<EntitySeed> visitedSeeds = new HashSet<>();
+    private int pathVertices;
+    private int hops;
+
+    @Override
+    public Iterable<Iterable<Edge>> doOperation(final Path operation, final Context context, final Store store) throws OperationException {
+
+        // Check input
+        if (null == operation.getInput() || !operation.getInput().iterator().hasNext()) {
+            return null;
+        }
+
+        // Check operations input
+        final Iterable<GetElements> operations = operation.getOperations();
+
+        if (null == operations || Iterables.isEmpty(operations)) {
+            return null;
+        }
+
+        // Validate the View objects
+        for (final GetElements op : operations) {
+            if (null != op.getView() && op.getView().hasEntities()) {
+                throw new OperationException("The view for operation " + op + " must not contain Entities.");
+            }
+        }
+
+        Iterable<? extends EntitySeed> seeds = operation.getInput();
+
+        hops = Iterables.size(operation.getOperations());
+        pathVertices = hops + 1;
+
+        final HashBasedTable<Object, Object, Set<Edge>> edgeTable = HashBasedTable.create();
+
+        // Execute the GetElements operations
+        for (final GetElements op : operations) {
+
+            final Set<EntitySeed> opSeeds = difference(visitedSeeds, Sets.newHashSet(seeds));
+
+            op.setInput(opSeeds);
+            final Iterable<? extends Element> results = store.execute(op, context);
+            visitedSeeds.addAll(opSeeds);
+
+            // Cache results
+            Lists.newArrayList(results).stream()
+                    .map(e -> (Edge) e)
+                    .forEach(e -> {
+                        if (null != edgeTable.get(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue())) {
+                            final Set<Edge> set = edgeTable.get(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue());
+                            set.add(e);
+                            edgeTable.put(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue(), set);
+                        } else {
+                            final Set<Edge> set = new HashSet<>();
+                            set.add(e);
+                            edgeTable.put(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue(), set);
+                        }
+                    });
+
+            final OperationChain<Set<? extends EntitySeed>> opChain = new OperationChain.Builder()
+                    .first(new ToVertices.Builder()
+                            .input(results)
+                            .edgeVertices(ToVertices.EdgeVertices.DESTINATION)
+                            .build())
+                    .then(new ToEntitySeeds())
```

Comment:
Perhaps we could skip the ToEntitySeeds operation here and just convert them to entity seeds as we execute the next operation (if there is even a next operation)